contracts/common/lib/BytesLib.sol

```diff
@@ -134,11 +135,13 @@ library BytesLib {
 
   // Pad a bytes array to 32 bytes
   function leftPad(bytes memory _bytes) internal pure returns (bytes memory) {
-    bytes memory newBytes = new bytes(32 - _bytes.length);
+    // may underflow if bytes.length < 32. Hence using SafeMath.sub
+    bytes memory newBytes = new bytes(SafeMath.sub(32, _bytes.length));
     return concat(newBytes, _bytes);
   }
 
   function toBytes32(bytes memory b) internal pure returns (bytes32) {
+    require(b.length >= 32, "Bytes array should atleast be 32 bytes");
```

Comment:
we should have one function without this check because there might be a case where we might need to convert the smaller data type to bytes32.

Reply:
Can we add that function when we actually require it?

Reply:
yeah sure works for me.